qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java

```diff
@@ -0,0 +1,284 @@
+    }
+
+    @Then("the system loads the manage location attribute type page")
+    public void theSystemLoadsTheManageLocationAttributeTypePage() {
+        assertTrue(textExists("Manage Location Attribute Types"));
+    }
+
+    @And("a user clicks on add new location attribute type")
+    public void userClicksOnAddNewLocationAttributeType() {
+        manageLocationAttributeTypesPage.goToAddNewLocationAttributeTypeForm();
+    }
+
+    @And("a user fills the add new location attribute type form")
+    public void userFillsTheAddNewLocationAttributeTypeForm() {
+        manageLocationAttributeTypesPage.enterLocationAttributeTypeName(LOCATION_ATTRIBUTE_TYPE_NAME);
+        manageLocationAttributeTypesPage.enterLocationAttributeTypeDescription(LOCATION_ATTRIBUTE_TYPE_DESCRIPTION);
+        manageLocationAttributeTypesPage.enterMinimumOccurs(MINIMUM_OCCURS);
+        manageLocationAttributeTypesPage.enterMaximumOccurs(MAXIMUM_OCCURS);
+        manageLocationAttributeTypesPage.selectDatatype("Location Datatype");
+        manageLocationAttributeTypesPage.enterDatatypeConfiguration(DATATYPE_CONFIGURATION);
+        manageLocationAttributeTypesPage.selectPreferredHandler("Location Field Gen Datatype Handler");
+        manageLocationAttributeTypesPage.enterHandlerConfiguration(LOCATION_HANDLER_CONFIGURATION);
+    }
+
+    @Then("the user saves the form")
+    public void userSavesTheForm() {
+        manageLocationAttributeTypesPage.clickOnSaveButton();
+        // assertTrue(textExists("Location Attribute Type Saved"));
+    }
+
+    @And("the user clicks on the edit icon")
+    public void userClicksOnTheEditIcon() {
+        editLocationAttributeTypeFormPage = manageLocationAttributeTypesPage.goToEditLocationAttributeTypeForm();
+    }
+
+    @And("the user fills in the prefered details in the edit location attribute type form")
+    public void userFillsInPreferedDetails() {
+        editLocationAttributeTypeFormPage.editLocationAttributeTypeName(LOCATION_ATTRIBUTE_TYPE_NAME);
+        editLocationAttributeTypeFormPage
+                .editLocationAttributeTypeDescription("editedLocationAttributeTypeDescription");
+        editLocationAttributeTypeFormPage.editedMaximumOccurs("1000");
+    }
+
+    @Then("the user saves the edit location attribute type form")
+    public void saveEditLocationAttributeTypeForm() {
+        editLocationAttributeTypeFormPage.saveEditForm();
+         assertTrue(textExists("Location Attribute Type Saved"));
+    }
+
+    @And("the user clicks on the retire icon")
+    public void userClicksOnTheRetireIcon() {
+        manageLocationAttributeTypesPage.retireLocation();
+    }
+
+    @Then("the system retires the location attribute type")
+    public void systemRetiresTheLocationAttributeType() {
+        assertTrue(textExists("Location Attribute Type Retired"));
+    }
+
+    @And("the user clicks the restore location attribute type icon")
```

Comment:
```suggestion
    @And("the user clicks  on the restore location attribute type icon")
```

Reply:
Could you review those feature files have start with `user click the to user clicks on `, it sounds so better.
This looks so nice to be merged. few comments to address.

Reply:
alright, thanks